nomad/client_agent_endpoint.go

```diff
@@ -402,54 +375,126 @@ func (a *Agent) forwardMonitorServer(conn io.ReadWriteCloser, server *serverPart
 	structs.Bridge(conn, serverConn)
 	return
 }
-
 func (a *Agent) forwardProfileClient(args *structs.AgentPprofRequest, reply *structs.AgentPprofResponse) error {
-	nodeID := args.NodeID
+	state, srv, err := a.findClientConn(args.NodeID)
+
+	if err != nil {
+		return err
+	}
+
+	if srv != nil {
+		return a.srv.forwardServer(srv, "Agent.Profile", args, reply)
+	}
+
+	// NodeRpc
+	rpcErr := NodeRpc(state.Session, "Agent.Profile", args, reply)
+	if rpcErr != nil {
+		return rpcErr
+	}
+
+	return nil
+}
+
+// Host returns data about the agent's host system for the `debug` command.
+func (a *Agent) Host(args *structs.HostDataRequest, reply *structs.HostDataResponse) error {
+
+	aclObj, err := a.srv.ResolveToken(args.AuthToken)
+	if err != nil {
+		return err
+	} else if aclObj != nil && !aclObj.AllowAgentRead() {
+		return structs.ErrPermissionDenied
+	}
+
```

Comment:
not sure exactly how sensitive all the host data is, but we do the following for profile endpoints, may be worth adding as well, 
```go

	// If ACLs are disabled, EnableDebug must be enabled
	if aclObj == nil && !a.srv.config.EnableDebug {
		return structs.ErrPermissionDenied
	}
```

Reply:
Fixed, for all the ACL endpoints